libs/service/api_service.cpp

```diff
@@ -205,22 +205,28 @@ _query_and_cache_map_descriptors(fd_handle_map* handle_map, uint32_t handle_map_
         for (uint32_t i = 0; i < handle_map_count; i++) {
             uint32_t size;
             descriptor = {0};
+            uint32_t inner_map_idx;
             result = query_map_definition(
                 handle_map[i].handle,
                 &size,
                 &descriptor.type,
                 &descriptor.key_size,
                 &descriptor.value_size,
-                &descriptor.max_entries);
+                &descriptor.max_entries,
+                &inner_map_idx);
             if (result != EBPF_SUCCESS) {
                 return result;
             }
 
+            // Convert inner_map_idx to descriptor.inner_map_fd.
+            descriptor.inner_map_fd = handle_map[inner_map_idx].file_descriptor;
```

Comment:
Should we populate this only when inner_map_idx is > 0?

For a case where there are 3 regular maps, inner_map_idx for each of them will be 0, but we will fill the file_descriptor of map[0] as inner_map_fd for all the 3 maps.

Reply:
0 is a valid index.  Instead, I wrapped this in a check for the map type not being a regular map